src/better-bem.ts

```diff
@@ -15,22 +31,22 @@ const GLUE_PROP = '-';
 /**
  * cleanClassNamesArray
  *
- * @param {string|array|Object} input               classnames which should be cleaned
- * @param {Boolean}             useKeyValuePairs    if true handle prop-value mod classnames
+ * @param input               classnames which should be cleaned
+ * @param useKeyValuePairs    if true handle prop-value mod classnames
  *
- * @return {array} clean array clean classnames
+ * @return clean array clean classnames
  */
-const cleanClassNamesArray = (input = [], useKeyValuePairs = false, keyValueGlue) => (
+const cleanClassNamesArray = (input: ClassNames = [], useKeyValuePairs = false, keyValueGlue = ''): string[] => (
     // make sure input is an array
     flatten([input])
         // reduce into new array with classname strings
-        .reduce((acc, classNames) => {
+        .reduce<string[]>((acc, classNames) => {
             // filter for strings only
-            if (isString(classNames)) {
+            if (isString(classNames) && typeof classNames == 'string') {
```

Comment:
Although `isString()` does the same check, the `typeof` check must be explicit to give type hints to TypeScript compiler.